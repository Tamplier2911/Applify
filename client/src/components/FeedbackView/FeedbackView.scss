/*

.feedback {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 2rem;

  padding: 2rem;
  align-items: center;

  @media only screen and (max-width: 678px) {
    grid-template-columns: 1fr;
  }

  &__left {
    &--container {
      position: relative;
      display: grid;
    }

    &--imageWrap {
      grid-row: 1 / -1;
      grid-column: 1 / -1;
      order: 2;

      width: 14rem;
      height: 14rem;
      border: 0.3rem solid var(--cl-primary);
      border-radius: 50%;
      overflow: hidden;

      box-shadow: 0 0.2rem 0.2rem rgba(0, 0, 0, 0.459);

      @media only screen and (max-width: 310px) {
        justify-self: center;
        margin-bottom: 1rem;

        width: 12rem;
        height: 12rem;
      }
    }

    &--img {
      width: 100%;
      height: 100%;
      display: block;
      object-fit: cover;
    }

    &--ratingWrap {
      grid-row: 1 / -1;
      grid-column: 1 / -1;
      order: 1;

      align-self: center;

      display: grid;
      grid-template-columns: repeat(5, 1fr);
      justify-items: center;

      padding: 1.6rem;

      background-image: linear-gradient(
        to right,
        transparent 20%,
        var(--cl-primary) 20%
      );

      @media only screen and (max-width: 876px) {
        grid-template-columns: repeat(3, 1fr);
      }

      @media only screen and (max-width: 678px) {
        grid-template-columns: repeat(5, 1fr);
        background-image: linear-gradient(
          to right,
          transparent 10%,
          var(--cl-primary) 10%
        );
      }

      @media only screen and (max-width: 360px) {
        grid-template-columns: repeat(3, 1fr);
      }

      @media only screen and (max-width: 310px) {
        grid-template-columns: 1fr;
        grid-row: 2 / 3;
        background-image: linear-gradient(
          to right,
          var(--cl-primary),
          var(--cl-primary)
        );
      }
    }

    &--name {
      font-family: var(--font-title);
      grid-column: 2 / -1;
      font-size: 2rem;
      color: var(--cl-white);

      width: 15rem;
      text-align: center;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;

      @media only screen and (max-width: 768px) {
        width: 14rem;
      }

      @media only screen and (max-width: 678px) {
        width: 20rem;
      }

      @media only screen and (max-width: 425px) {
        grid-column: 5 / 6;
      }

      @media only screen and (max-width: 360px) {
        width: 15rem;
      }

      @media only screen and (max-width: 310px) {
        font-size: 1.8rem;
        grid-column: 1 / -1;
      }
    }

    &--rating {
      grid-column: 2 / -1;
      display: grid;
      grid-template-columns: repeat(5, min-content);
      grid-column-gap: 1rem;
      justify-self: center;

      @media only screen and (max-width: 425px) {
        grid-column: 5 / 6;
      }

      @media only screen and (max-width: 310px) {
        grid-column: 1 / -1;
      }
    }

    &--svg {
      width: 2rem;
      height: 2rem;

      @media only screen and (max-width: 310px) {
        width: 1.5rem;
        height: 1.5rem;
      }
    }
  }

  &__right {
    &--container {
      display: grid;
      grid-template-columns: min-content 1fr min-content;
      grid-column-gap: 1rem;
    }

    &--qOpen {
      align-self: start;
      font-size: 6rem;
      color: var(--cl-primary);
    }

    &--text {
      align-self: center;
      font-family: var(--font-title);
      font-size: 1.8rem;
      padding: 1rem;
      font-style: italic;
    }

    &--qClose {
      align-self: end;
      font-size: 6rem;
      color: var(--cl-primary);
    }
  }
}

// left and right
.order-big {
  order: 2;

  @media only screen and (max-width: 678px) {
    order: 0;
  }
}

.order-small {
  order: 0;
}

// image wrapper
.img-wrap-even {
  justify-self: end;

  @media only screen and (max-width: 310px) {
    justify-self: center;
  }
}

// content wrapper
.cnt-wrap-even {
  grid-template-columns: repeat(5, 1fr);
  background-image: linear-gradient(
    to right,
    var(--cl-primary) 90%,
    transparent 90%
  );

  @media only screen and (max-width: 876px) {
    grid-template-columns: repeat(5, 1fr);
  }

  @media only screen and (max-width: 678px) {
    grid-template-columns: repeat(5, 1fr);
  }

  @media only screen and (max-width: 360px) {
    grid-template-columns: repeat(5, 1fr);
  }

  @media only screen and (max-width: 310px) {
    background-image: linear-gradient(
      to right,
      var(--cl-primary),
      var(--cl-primary)
    );
  }
}

// content positions
.cnt-name-even {
  grid-column: 1 / 5;

  @media only screen and (max-width: 425px) {
    grid-column: 1 / 2;
  }

  @media only screen and (max-width: 310px) {
    grid-column: 1 / -1;
  }
}

.cnt-rating-even {
  grid-column: 1 / 5;

  @media only screen and (max-width: 425px) {
    grid-column: 1 / 2;
  }

  @media only screen and (max-width: 310px) {
    grid-column: 1 / -1;
  }
}

// triangle
.triangle-odd {
  position: absolute;
  right: -1.4rem;
  bottom: 36%;

  width: 0;
  height: 0;
  border-top: 2rem solid transparent;
  border-bottom: 2rem solid transparent;

  border-left: 2rem solid var(--cl-primary);

  @media only screen and (max-width: 678px) {
    display: none;
  }
}

.triangle-even {
  position: absolute;
  left: -1.4rem;
  bottom: 36%;

  width: 0;
  height: 0;
  border-top: 2rem solid transparent;
  border-bottom: 2rem solid transparent;

  border-right: 2rem solid var(--cl-primary);

  @media only screen and (max-width: 678px) {
    display: none;
  }
}

*/
