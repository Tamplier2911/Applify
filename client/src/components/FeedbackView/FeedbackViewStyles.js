import styled, { css } from "styled-components";

// svg
import { ReactComponent as StarSVG } from "../../assets/svg/star.svg";

const getQuoteColor = props => {
  const { color } = props;
  return `color ${color};`;
};

const getOrder = props => {
  const { even } = props;
  if (even) {
    return `order: 2;
            @media only screen and (max-width: 678px) {
                order: 0;
            }`;
  } else {
    return `order: 0;`;
  }
};

const ImageWrapSharedStyles = css`
  grid-row: 1 / -1;
  grid-column: 1 / -1;
  order: 2;
  width: 14rem;
  height: 14rem;
  border-radius: 50%;
  overflow: hidden;
  box-shadow: 0 0.2rem 0.2rem var(--cl-feedback-view-shadow);

  @media only screen and (max-width: 310px) {
    justify-self: center;
    margin-bottom: 1rem;
    width: 12rem;
    height: 12rem;
  }
`;

const getImageWrapStyles = props => {
  const { even, color } = props;
  if (even) {
    return `border: 0.3rem solid ${color};
            background-color: ${color};
            justify-self: end;`;
  } else {
    return `border: 0.3rem solid ${color};
            background-color: ${color};`;
  }
};

const RatingWrapSharedStyles = css`
  grid-row: 1 / -1;
  grid-column: 1 / -1;
  order: 1;
  align-self: center;
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  justify-items: center;
  padding: 1.6rem;

  @media only screen and (max-width: 310px) {
    grid-template-columns: 1fr;
    grid-row: 2 / 3;
  }
`;

const getRatingWrapStyles = props => {
  const { even, color } = props;
  if (even) {
    return `grid-template-columns: repeat(5, 1fr);
            background-image: linear-gradient(
                to right,
                ${color} 90%,
                transparent 90%
            );
        
            @media only screen and (max-width: 876px) {
                grid-template-columns: repeat(5, 1fr);
            }
        
            @media only screen and (max-width: 678px) {
                grid-template-columns: repeat(5, 1fr);
            }
        
            @media only screen and (max-width: 360px) {
                grid-template-columns: repeat(5, 1fr);
            }
        
            @media only screen and (max-width: 310px) {
                background-image: linear-gradient(
                    to right,
                    ${color},
                    ${color}
                );
            }`;
  } else {
    return `background-image: linear-gradient(
                to right,
                transparent 10%,
                ${color} 10%
            );

            @media only screen and (max-width: 876px) {
                grid-template-columns: repeat(3, 1fr);
            }

            @media only screen and (max-width: 678px) {
                grid-template-columns: repeat(5, 1fr);
            }

            @media only screen and (max-width: 360px) {
                grid-template-columns: repeat(3, 1fr);
            }

            @media only screen and (max-width: 310px) {
                background-image: linear-gradient(
                  to right,
                  ${color},
                  ${color}
                );
            }
    `;
  }
};

const FeedbackNameSharedStyles = css`
  font-family: var(--font-title);
  font-size: 2rem;
  color: var(--cl-white);
  width: 15rem;
  text-align: center;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;

  @media only screen and (max-width: 768px) {
    width: 14rem;
  }

  @media only screen and (max-width: 678px) {
    width: 20rem;
  }

  @media only screen and (max-width: 360px) {
    width: 15rem;
  }

  @media only screen and (max-width: 310px) {
    font-size: 1.8rem;
    grid-column: 1 / -1;
  }
`;

const getFeedbackNameStyles = props => {
  const { even } = props;
  if (even) {
    return `grid-column: 1 / 5;
            @media only screen and (max-width: 425px) {
                grid-column: 1 / 2;
            }`;
  } else {
    return `grid-column: 2 / -1;
            @media only screen and (max-width: 425px) {
                grid-column: 5 / 6;
            }`;
  }
};

const FeedbackRatingSharedStyles = css`
  display: grid;
  grid-template-columns: repeat(5, min-content);
  grid-column-gap: 1rem;
  justify-self: center;
`;

const getFeedbackRatingStyles = props => {
  const { even } = props;
  if (even) {
    return `grid-column: 1 / 5;
            @media only screen and (max-width: 425px) {
                grid-column: 1 / 2;
            }
            @media only screen and (max-width: 310px) {
                grid-column: 1 / -1;
            }`;
  } else {
    return `grid-column: 2 / -1;
            @media only screen and (max-width: 425px) {
                grid-column: 5 / 6;
            }
            @media only screen and (max-width: 310px) {
                grid-column: 1 / -1;
            }`;
  }
};

const TriangleSharedStyles = css`
  position: absolute;
  bottom: 36%;
  width: 0;
  height: 0;
  border-top: 2rem solid transparent;
  border-bottom: 2rem solid transparent;

  @media only screen and (max-width: 678px) {
    display: none;
  }
`;

const getTriangleStyles = props => {
  const { even, color } = props;
  if (even) {
    return `
    left: -1.4rem;
    border-right: 2rem solid ${color};`;
  } else {
    return `
    right: -1.4rem;
    border-left: 2rem solid ${color};`;
  }
};

export const FeedbackViewContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 2rem;

  padding: 2rem;
  align-items: center;

  @media only screen and (max-width: 678px) {
    grid-template-columns: 1fr;
  }
`;

export const FeedbackViewLeft = styled.div`
  ${getOrder}
`;

export const FeedbackViewLeftCont = styled.div`
  position: relative;
  display: grid;
`;

export const FeedbackViewLeftImgWrap = styled.div`
  ${ImageWrapSharedStyles}
  ${getImageWrapStyles}
`;

export const FeedbackViewLeftImg = styled.img`
  width: 100%;
  height: 100%;
  display: block;
  object-fit: cover;
`;

export const FeedbackViewLeftRatingWrap = styled.div`
  ${RatingWrapSharedStyles}
  ${getRatingWrapStyles}
`;

export const FeedbackViewLeftName = styled.div`
  ${getFeedbackNameStyles}
  ${FeedbackNameSharedStyles}
`;

export const FeedbackViewLeftRating = styled.div`
  ${getFeedbackRatingStyles}
  ${FeedbackRatingSharedStyles}
`;

export const FeedbackViewLeftSVG = styled(StarSVG)`
  width: 2rem;
  height: 2rem;

  @media only screen and (max-width: 310px) {
    width: 1.5rem;
    height: 1.5rem;
  }
`;

export const FeedbackViewLeftTriangle = styled.div`
  ${TriangleSharedStyles}
  ${getTriangleStyles}
`;

export const FeedbackViewRight = styled.div``;

export const FeedbackViewRightCont = styled.div`
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  grid-column-gap: 1rem;
`;

export const FeedbackViewRightQLeft = styled.div`
  ${getQuoteColor}
  align-self: start;
  font-size: 6rem;
`;

export const FeedbackViewRightText = styled.div`
  align-self: center;
  font-family: var(--font-title);
  font-size: 1.8rem;
  padding: 1rem;
  font-style: italic;
`;

export const FeedbackViewRightQRight = styled.div`
  ${getQuoteColor}
  align-self: end;
  font-size: 6rem;
`;
