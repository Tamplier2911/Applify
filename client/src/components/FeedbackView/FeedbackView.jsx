import "./FeedbackView.scss";
import React from "react";

// svg
import { ReactComponent as StarSVG } from "../../assets/svg/star.svg";

// JS Rendering CSS
import {} from "./FeedbackViewStyles";

export const FeedbackView = ({
  name,
  photo,
  rating,
  feedback,
  count,
  color
}) => {
  const even = count % 2 === 0 ? 1 : 0;
  const groundedRating = Math.floor(rating);

  return (
    <div className="feedback">
      <div className={`feedback__left ${even ? "order-big" : "order-small"}`}>
        <div className="feedback__left--container">
          <div
            style={{ border: `0.3rem solid ${color}` }}
            className={`feedback__left--imageWrap ${
              even ? "img-wrap-even" : ""
            }`}
          >
            <img
              className="feedback__left--img"
              alt="happy user"
              src={photo}
            ></img>
          </div>
          <div
            style={
              even
                ? {
                    backgroundImage: `linear-gradient(to right, ${color} 90%, transparent 90%)`
                  }
                : {
                    backgroundImage: `linear-gradient(to right, transparent 10%, ${color} 10%)`
                  }
            }
            className={`feedback__left--ratingWrap ${
              even ? "cnt-wrap-even" : ""
            }`}
          >
            <div
              className={`feedback__left--name ${even ? "cnt-name-even" : ""}`}
            >
              {name}
            </div>
            <div
              className={`feedback__left--rating ${
                even ? "cnt-rating-even" : ""
              }`}
            >
              {Array.from(new Array(groundedRating), (el, i) => i + 1).map(
                el => {
                  return <StarSVG key={el} className="feedback__left--svg" />;
                }
              )}
            </div>
          </div>
          <div
            style={
              even
                ? { borderRight: `${color} 2rem solid` }
                : { borderLeft: `${color} 2rem solid` }
            }
            className={`${even ? "triangle-even" : "triangle-odd"}`}
          ></div>
        </div>
      </div>
      <div className="feedback__right">
        <div className="feedback__right--container">
          <div style={{ color: `${color}` }} className="feedback__right--qOpen">
            &ldquo;
          </div>
          <div className="feedback__right--text">{feedback}</div>
          <div
            style={{ color: `${color}` }}
            className="feedback__right--qClose"
          >
            &bdquo;
          </div>
        </div>
      </div>
    </div>
  );
};

export default FeedbackView;
