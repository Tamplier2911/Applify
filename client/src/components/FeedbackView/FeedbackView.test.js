import React from "react";
import toJson from "enzyme-to-json";

// component
import FeedbackView from "./FeedbackView";

// utils
import { setShallow, findByTestAttr, checkProps } from "../../tests/testUtils";

describe("<FeedbackView />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallow(<FeedbackView user={{}} rating={"2"} />);
  });

  it("renders without an error", () => {
    const feedbackView = findByTestAttr(wrapper, "feedback-view");
    expect(feedbackView.length).toBe(1);
  });

  it("renders children - left", () => {
    const feedbackViewLeft = findByTestAttr(wrapper, "feedback-view-left");
    expect(feedbackViewLeft.length).toBe(1);
  });

  it("renders children - right", () => {
    const feedbackViewRight = findByTestAttr(wrapper, "feedback-view-right");
    expect(feedbackViewRight.length).toBe(1);
  });

  it("recieves expected props", () => {
    const expectedProps = {
      count: 2,
      rating: "5",
      feedback: "Good!",
      color: "#333",
      user: {},
    };
    const err = checkProps(FeedbackView, expectedProps, "props");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const feedbackView = findByTestAttr(wrapper, "feedback-view");
    expect(toJson(feedbackView)).toMatchSnapshot();
  });
});
