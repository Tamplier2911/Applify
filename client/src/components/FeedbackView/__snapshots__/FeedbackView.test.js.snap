// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackView /> matches snapshot 1`] = `ShallowWrapper {}`;
